homeassistant/components/roborock/roborock_storage.py

```diff
@@ -48,9 +50,20 @@ def _load_map(self, filename: Path) -> bytes | None:
             return None
 
     async def async_save_map(self, map_flag: int, content: bytes) -> None:
-        """Write map if it should be updated."""
-        filename = self._path_prefix / f"{map_flag}{MAP_FILENAME_SUFFIX}"
-        await self._hass.async_add_executor_job(self._save_map, filename, content)
+        """Save the map to a pending write queue."""
+        self._write_queue[map_flag] = content
+
+    async def flush(self) -> None:
+        """Flush all maps to disk."""
+        tasks = []
+        _LOGGER.debug("Flushing %s maps to disk", len(self._write_queue))
+        for map_flag, content in self._write_queue.items():
+            filename = self._path_prefix / f"{map_flag}{MAP_FILENAME_SUFFIX}"
+            tasks.append(
+                self._hass.async_add_executor_job(self._save_map, filename, content)
```

Comment:
Doesn't this spawn a lot (n = size of `self._write_queue`) of executor jobs?

Reply:
Updated. Before if you have 20 robot vacuums in your home it will spawn 20 executor jobs.  Now it will  spawn one executor and block on 20 sequential writes.